pkg/service/providerindex/providerindex.go

```diff
@@ -175,6 +186,17 @@ func (pi *ProviderIndexService) cacheResults(ctx context.Context, s trace.Span,
 	return results, nil
 }
 
+// Helper function to cache empty results.
+func (pi *ProviderIndexService) cacheNoProviderResults(ctx context.Context, s trace.Span, mh mh.Multihash) ([]model.ProviderResult, error) {
+	s.AddEvent("caching results")
+	err := pi.noProviderStore.Set(ctx, mh, struct{}{}, true)
```

Comment:
for some reason, this is setting the expiration time to the 1 hour default rather than the configured time. As it is today, you need to do a `Set` and then a `SetExpirable`.

If nobody remembers why this is this way, IMO we should just change `Set` so that it sets the configured expiration time when setting the key. Can be done in a different PR.

Reply:
wow no there's a bunch. including a fix -- thanks for catching that.